client_api/data_access/gumbo_data_access.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+import pandas as pd
+
+
+class GumboDAO:
+    """Data Access Object for """
+    def __init__(
+        self,
+        psycopg2_connection,
+        sanity_check=True,
+        username=None,
+        autocommit=True
+    ):
+        self.sanity_check = sanity_check
+        self.connection = psycopg2_connection
+        self.connection.autocommit = autocommit
+
+        # set the username for use in audit logs
+        self.username = username or os.getlogin() + " (py)"
+        with self.connection.cursor() as cursor:
+            print("setting username to", self.username)
+            cursor.execute("SET my.username=%s", [self.username])
+
+
+    def get(self, table_name):
+        cursor = self.connection.cursor()
+        select_query = f"select * from {table_name}"
+
+        # If a primary key exists, use it to sort the table
+        # Views don't have primary keys to use here, and that's fine.
+        try:
+            pk_column = _get_pk_column(cursor, table_name)
+            select_query += f" order by {pk_column}"
+        except AssertionError:
```

Comment:
What do you think of changing the check to throw a custom exception? (because we're catching it here)